core/string/ustring.cpp

```diff
@@ -4246,6 +4246,64 @@ String String::replace_first(const char *p_key, const char *p_with) const {
 	return *this;
 }
 
+String String::replace_char(char32_t p_key, char32_t p_with) const {
+	ERR_FAIL_COND_V_MSG(p_with == 0, String(), "`with` must not be null.");
+
+	if (p_key == 0 || is_empty()) {
+		return *this;
+	}
+
+	String new_string;
+
+	new_string.resize(size());
+
+	char32_t *new_ptrw = new_string.ptrw();
+	const char32_t *old_ptr = ptr();
+
+	while (*old_ptr) {
+		if (*old_ptr == p_key) {
+			*new_ptrw = p_with;
+		} else {
+			*new_ptrw = *old_ptr;
+		}
+		++new_ptrw;
+		++old_ptr;
+	}
+
+	*new_ptrw = 0;
+
+	return new_string;
+}
+
+String String::replace_chars(const Vector<char32_t> &p_keys, char32_t p_with) const {
+	ERR_FAIL_COND_V_MSG(p_with == 0, String(), "`with` must not be null.");
+
+	if (p_keys.is_empty() || is_empty()) {
+		return *this;
+	}
+
+	String new_string;
+
+	new_string.resize(size());
+
+	char32_t *new_ptrw = new_string.ptrw();
+	const char32_t *old_ptr = ptr();
+
+	while (*old_ptr) {
+		if (p_keys.has(*old_ptr)) {
```

Comment:
Could have the keys be copied and then sorted here and use `bsearch` instead but not sure it'll be worth it generally, not expecting large arrays here so the difference between `O(n)` and `O(log n)` will be marginal compared to the copying and sorting